### ISSUE TITLE
Run accessibility audit

### ISSUE THREAD

**Issue by gemfarmer** [ONRR/doi-extractives-data#2252]
There are several pages that are less accessible than ideal. Run accessibility audit on all pages and make changes accordingly

**Comment by gemfarmer** [comment]
@toolness this would be great to have your expertise on. We were originally running `pa11y-ci`, but we forgot for some reason that I don't remember now

**Comment by jennmalcolm** [comment]
We have some accessibility issues with charts especially, so we need to run this audit. Our first step is to set the accessibility level we want to achieve.

**Comment by jennmalcolm** [comment]
We are reaching for WCAG 2.0 AA

**Comment by brentryanjohnson** [comment]
Resources for accessibility (including audit tools):

- [18F accessibility guide](https://accessibility.18f.gov/)
- [Accessibility for teams](https://accessibility.digital.gov/)

**Comment by brentryanjohnson** [comment]
I ran two separate accessibility audits on revenuedata.doi.gov/explore/. 

- [Pa11y command-line audit](https://github.com/pa11y/pa11y) using HTMLCodeSniffer
- [Chrome developer tools audits (Lighthouse)](https://developers.google.com/web/tools/lighthouse/)

## Pa11y errors

### Search
```
 • Error: This search input element does not have a name available to an accessibility API. Valid names are: label eleme
nt, title attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputSearch.Name
   ├── #search-input
   └── <input type="search" class="search-box header-nav_search" placeholder="Search" id="search-input" name="q" role="s
earch">
```

### Map (applies to each state and offshore area)
#### Alaska example
```
 • Error: Anchor element found with a valid href attribute, but no link content has been supplied.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.A.NoContent
   ├── #title > div > div:nth-child(1) > figure > div > svg > a:nth-child(2)
   └── <a xlink:href="/explore/offshore-alaska/" href="/explore/offshore-alaska/"> <g class="state offshore-a...</a>
```
### Select elements
```
 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, ti
tle attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #all-production > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, ti
tle attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #federal-production > section > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, ti
tle attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #revenue-trends > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, ti
tle attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #gdp > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, ti
tle attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #employment > section:nth-child(1) > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This select element does not have a name available to an accessibility API. Valid names are: label element, ti
tle attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Select.Name
   ├── #self-employment > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```

### Form field label
```
 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrappe
d around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #all-production > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrappe
d around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #federal-production > section > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrappe
d around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #revenue-trends > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrappe
d around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #gdp > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrappe
d around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #employment > section:nth-child(1) > h3 > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```
```
 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrappe
d around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #self-employment > select
   └── <select class="chart-selector flex-row-icon"> ...</select>
```

### Duplicate id attributes
```
 • Error: Duplicate id attribute value "federal-production" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #federal-production
   └── <section id="federal-production" is="year-switcher-section" class="federal production"> <section class="county-ma
p-...</section>
```
```
 • Error: Duplicate id attribute value "Resources-to-Revenue" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #Resources-to-Revenue
   └── <g id="Resources-to-Revenue"><g id="_1.0-Resources-to-Revenu...</g>
```
```
 • Error: Duplicate id attribute value "_1.0-Resources-to-Revenue-Landing" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #_1.0-Resources-to-Revenue-Landing
   └── <g id="_1.0-Resources-to-Revenue-Landing" data-name="1.0-Resources-to-Revenue-Landing"><g id="panel-6-copy"><g id
="Ico...</g>
```
```
 • Error: Duplicate id attribute value "panel-6-copy" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #panel-6-copy
   └── <g id="panel-6-copy"><g id="Icon:-Coal"><path d="M24...</g>
```
```
 • Error: Duplicate id attribute value "Icon:-Coal" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #Icon:-Coal
   └── <g id="Icon:-Coal"><path d="M24.05,40.18a4.38,4.38...</g>
```
```
 • Error: Duplicate id attribute value "Fill-4" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #Fill-4
   └── <path id="Fill-4" class="cls-1" d="M34,42.9H31.53a1.64,1.64,0,1,0,0,3.27H34a1.64,1.64,0,1,0,0-3.27" transform="tr
anslate(-0.48 -0.23)"></path>
```

### Table elements
```
 • Error: Not all th elements in this table have a scope attribute. These cells should contain a scope attribute to iden
tify their association with td elements.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.H63.1
   ├── #revenue-types
   └── <table is="bar-chart-table" class="revenue table-arrow_box" id="revenue-types"> <thead> <tr> <th c...</table>
```

## Chrome developer tools audit
We can download a `json` formatted report, but the GUI wouldn't let me copy/paste out (thus the screenshots).

### ARIA attributes
- `[aria-*] attributes are not valid or misspelled`
![aria attribute error button element](https://user-images.githubusercontent.com/32855580/44740494-7a245400-aaaf-11e8-803a-b2e79141d357.png)

### Form elements
- Form elements do not have associated labels.
![Seven form elements lacking labels](https://user-images.githubusercontent.com/32855580/44740640-d2f3ec80-aaaf-11e8-927a-3ebc68192544.png)

### Color contrast
- Background and foreground colors do not have sufficient contrast ration
![color contrast audit result on class header-nav_item_link_spacer](https://user-images.githubusercontent.com/32855580/44740720-08003f00-aab0-11e8-92fd-2c033edc283a.png)

### Structured elements
- `id` attributes on the page are not unique
![id attributes are not unique, list of section, table, and svg elements](https://user-images.githubusercontent.com/32855580/44740810-354ced00-aab0-11e8-8cf1-2b4865dcaeaf.png)

- Lists do not contain only `li` elements and script supporting elements (`script` and `template`)
![unordered list element without list items](https://user-images.githubusercontent.com/32855580/44740865-5dd4e700-aab0-11e8-8076-c3ff09f0461a.png)

### Guidance for items to manually check

1. The page has a logical tab order
2. Interactive controls are keyboard focusable
3. The user's focus is directed to new content added to the page
4. User focus is not accidentally trapped in a region
5. Custom controls have associated labels
6. Custom controls have ARIA roles
7. Visual order on the page follows DOM order
8. Offscreen content is hidden from assistive technology
9. Headings don't skip levels
10. HTML5 landmark elements are used to improve navigation

**Comment by brentryanjohnson** [comment]
I'm closing this issue, and we can move on to fixing some of these errors.

@jennmalcolm Let me know if you want this audit list in a different format, and/or if we should import it into new issue(s) to start dealing with these errors.

**Comment by jennmalcolm** [comment]
I think this format is fine, but I would like to create new issues, divided as you see appropriate, for fixing the errors.
